src/js/techreport/combobox.js

```diff
@@ -0,0 +1,249 @@
+class ComboBox {
+  constructor(element, data) {
+    this.element = element;
+
+    this.data = data;
+    this.filteredData = data;
+    this.focusedOption = -1;
+    this.focusedOptionStr = '';
+    this.selected = [];
+    this.maxSelected = 10;
+
+    this.keyCodes = {
+      keyUp: 38,
+      keyDown: 40,
+      ESC: 27,
+      enter: 13,
+    };
```

Comment:
I'm not sure we need this? More explanation below.

Reply:
updated 👍🏻